SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp

```diff
@@ -817,6 +875,67 @@ void ASpatialFunctionalTest::OnReplicated_bFinishedTest()
 	}
 }
 
+void ASpatialFunctionalTest::LogAndClearRequireHandler()
+{
+	TArray<FSpatialFunctionalTestRequire> RequiresOrdered = RequireHandler.GetAndClearStepRequires();
+
+	ASpatialFunctionalTestFlowController* FlowController = GetLocalFlowController();
+	if (FlowController)
+	{
+		const FString& WorkerName = FlowController->GetDisplayName();
+
+		for (const auto& Require : RequiresOrdered)
+		{
+			FString Msg;
+			if (Require.bPassed)
+			{
+				Msg = FString::Printf(TEXT("%s [Passed] %s : %s"), *WorkerName, *Require.Msg, *Require.StatusMsg);
+			}
+			else
+			{
+				Msg = FString::Printf(TEXT("%s [Failed] %s : %s"), *WorkerName, *Require.Msg, *Require.StatusMsg);
+			}
+
+			FlowController->ServerNotifyLogRequireMessages(*Msg, Require.bPassed);
+		}
+	}
+}
+
+void ASpatialFunctionalTest::LogRequireMessages(const FString& Message, bool bPassed)
+{
+	if (bPassed)
+	{
+		UE_VLOG(nullptr, LogSpatialGDKFunctionalTests, Display, TEXT("%s"), *Message);
+		UE_LOG(LogSpatialGDKFunctionalTests, Display, TEXT("%s"), *Message);
+	}
+	else
+	{
+		UE_VLOG(nullptr, LogSpatialGDKFunctionalTests, Error, TEXT("%s"), *Message);
+		UE_LOG(LogSpatialGDKFunctionalTests, Error, TEXT("%s"), *Message);
+	}
```

Comment:
```suggestion
	UE_VLOG(nullptr, LogSpatialGDKFunctionalTests, (bPassed ? Display : Error), TEXT("%s"), *Message);
	UE_LOG(LogSpatialGDKFunctionalTests, (bPassed ? Display : Error), TEXT("%s"), *Message);
```

Reply:
Unfortunately this one causes compilation errors because of the way Verbosity is handled inside the UE_LOG macro. I've tried other approaches to shorten this like assigning the Verbosity to a variable as well but that also didn't work.

Reply:
ah yeah needs to be compile time constant because of macro usage my bad.